File: app/src/main/java/com/adityasubathu/vaccinationreminder/SplashActivity.java
package com.adityasubathu.vaccinationreminder;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

public class SplashActivity extends AppCompatActivity {

    public static String publicUsername;
    SharedPreferences mySharedPrefs, loginFlagPrefs;
    String activeUser;
    boolean loggedIn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        loginFlagPrefs = getSharedPreferences("loginInfo", MODE_PRIVATE);
        activeUser = loginFlagPrefs.getString("activeUser", null);
        loggedIn = loginFlagPrefs.getBoolean("loggedIn", false);

        new Thread() {

            public void run() {

                try {
                    Thread.sleep(2500);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }

                if (!loggedIn) {
                    Intent i = new Intent(SplashActivity.this, LoginActivity.class);
                    startActivity(i);
                    finish();

                    if (activeUser != null && !activeUser.isEmpty()) {
                        publicUsername = activeUser;
                    }
                } else {
                    LoginActivitySignUpFragment.activeUsername = activeUser;
                    Intent i2 = new Intent(SplashActivity.this, MainFragmentHolder.class);
                    startActivity(i2);
                    finish();
                }
            }
        }.start();
    }
}

File: app/src/main/java/com/adityasubathu/vaccinationreminder/SplashActivity.kt
package com.adityasubathu.vaccinationreminder

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class SplashActivity : AppCompatActivity() {

    private lateinit var loginFlagPrefs: SharedPreferences
    lateinit var activeUser: String
    var loggedIn = false

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        loginFlagPrefs = getSharedPreferences("loginInfo", Context.MODE_PRIVATE)
        activeUser = loginFlagPrefs.getString("activeUser", null).toString()
        loggedIn = loginFlagPrefs.getBoolean("loggedIn", false)

        object : Thread() {
            override fun run() {
                try {
                    sleep(2500)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }
                if (!loggedIn) {
                    val i = Intent(this@SplashActivity, LoginActivity::class.java)
                    startActivity(i)
                    finish()
                    if (activeUser.isNotEmpty()) {
                        publicUsername = activeUser
                    }
                } else {
                    LoginActivitySignUpFragment.activeUsername = activeUser
                    val i2 = Intent(this@SplashActivity, MainFragmentHolder::class.java)
                    startActivity(i2)
                    finish()
                }
            }
        }.start()
    }

    companion object {
        lateinit var publicUsername: String
    }
}